code/datums/elements/food/food_trash.dm

```diff
@@ -63,10 +60,12 @@
 	INVOKE_ASYNC(src, .proc/async_generate_trash, source)
 	qdel(source)
 
+
 /datum/element/food_trash/proc/open_trash(datum/source, mob/user)
 	SIGNAL_HANDLER
 
 	to_chat(user, "<span class='notice'>You open the [src]\'s shell, revealing \a [initial(trash.name)].</span>")
```

Comment:
The issue appears to be here.
[src] in this case is the /datum/element/food_trash, so it needs to take the parent of the element for a proper name.